### PR TITLE
Document single quote escape

### PR DESCRIPTION
**Pull request by willmostly**
## Description
This adds an example of how to escape single quotes in string literals.

## Release notes

( x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Review by nineinchnick — APPROVED** [review]

File: docs/src/main/sphinx/language/types.md [hunk 1]
```diff
@@ -155,6 +155,9 @@ before any Unicode character usage with 4 digits. In the examples above
 with 6 digits require usage of the plus symbol before the code. For example,
 you need to use `\+01F600` for a grinning face emoji.
 
+Single quotes in string literals can be escaped by using another single quote:
```
Comment (mosabua) [inline docs/src/main/sphinx/language/types.md:158]: While true as it stands.. I am not sure this is the correct spot in the docs .. it also applies to char at least ..

We also might want to mention using double quotes for identifiers somewhere .. unless we do already

Comment (nineinchnick) [reply docs/src/main/sphinx/language/types.md:158]: Identifiers were covered in https://github.com/trinodb/trino/pull/18181
When working on that we decided not to have a separate page for generic sql syntax and put all literals descriptions here in data types. I agree we might mention this in chars too, but that should be it.

Comment (nineinchnick) [reply docs/src/main/sphinx/language/types.md:158]: Oh well, maybe also in the query table function docs. Some people asked about it in the Trino slack, how to escape quotes around literals in query "parameters"

**Review by hashhar — APPROVED** [review]
While it's ok to discuss if there's a better place to move this information for now this is still useful and answers a question a lot of people have.

Merging.

**Pull request merged**